### DISCUSSION TITLE
Can we get the ast of il and a visitor?

### DISCUSSION THREAD

**Discussion by 0x410c** [icsharpcode/ILSpy#3161] [Development]
**Is your feature request related to a problem? Please describe.**
I am working on a lifter, instead of just iterating through instructions, i think having a il ast and using a visitor pattern over it will be a better approach. as ilspy uses internally an ilast, using that will be the best case scenario

**Describe the solution you'd like**
An api which can provide this functionality?

**Additional context**
None

**Comment by siegfriedpammer** [comment]
For future reference: https://github.com/icsharpcode/ILSpy/blob/cfb4f0f312c129c9518b8f6eea8bbff63b3176da/ICSharpCode.Decompiler/IL/Instructions.cs#L6821
